### PR TITLE
NC | NSFS | new_buckets_path should be optional

### PR DESCRIPTION
**Pull request by naveenpaul1**
### Explain the changes
1. new_buckets_path should be optional and set allow_bucket_creation to false if missing.

### Issues: Fixed #xxx / Gap #xxx
1. https://github.com/noobaa/noobaa-core/issues/7625

### Testing Instructions:
1. run account add without specifying new_buckets_path and try to create bucket using that account and verify the error

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/sdk/bucketspace_fs.js [hunk 1]
```diff
@@ -278,6 +276,9 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
 
     async create_bucket(params, sdk) {
         return bucket_semaphore.surround_key(String(params.name), async () => {
+            if (!sdk.requesting_account.allow_bucket_creation) {
+                throw new RpcError('BUCKET_CREATION_NOT_ALLOWED');
```
Comment (romayalon) [inline src/sdk/bucketspace_fs.js:280]: please change to 'throw new RpcError('UNAUTHORIZED', 'Not allowed to create new buckets');' 
found it in bucket_server :)

File: src/test/unit_tests/test_bucketspace_fs.js [hunk 2]
```diff
@@ -68,6 +69,7 @@ const account_user2 = {
     name: 'user2',
```
Comment (romayalon) [inline src/test/unit_tests/test_bucketspace_fs.js:69]: can you please add a test that checks a scenario where an account is not authorized to create a bucket?

Comment (naveenpaul1) [reply src/test/unit_tests/test_bucketspace_fs.js:69]: sure, will add that.

**Review by romayalon — APPROVED** [review]

**Comment by shirady** [comment]
@naveenpaul1 @romayalon it seems like a configuration change.
If needed it should be also reflected in the creation of the accounts for the Ceph test (currently in PR #7685, in the file [src/deploy/NVA_build/standalone_deploy_nsfs.sh](https://github.com/noobaa/noobaa-core/pull/7685/files#diff-71d8375cc7e11f9ced9a0a8aa2fced4b66ce67f4028748824530a8f3fd2c5c7c)).

**Comment by romayalon** [comment]
@shirady it's not needed, this makes new_buckets_path optional but the ceph accounts need new_buckets_path because they are creating buckets.

**Comment by liranmauda** [comment]
@naveenpaul1 as we talked in dev, tests should be written in jest
@nimrod-becker Should we block this till the tests are written in jest?

**Pull request merged**